src/test/ui/async-await/issue-62517.rs

```diff
@@ -0,0 +1,23 @@
+// Regression test for #62517. We used to ICE when you had an `async
+// fn` with an `impl Trait` return that mentioned a `dyn Bar` with no
+// explicit lifetime bound.
+//
+// edition:2018
+// build-pass
+
+#![feature(async_await)]
+
+trait FirstTrait {}
+trait SecondTrait {
+    type Item: ?Sized;
+}
+
+async fn foo(x: &str) -> impl SecondTrait<Item = dyn FirstTrait> {
```

Comment:
Can you also include the version with `-> impl Alpha<dyn Object>` for completeness (so we also cover type parameters and not just associated types)?

Reply:
Suggestion to apply in https://github.com/rust-lang/rust/pull/63376#discussion_r311926986